src/tools/clippy/tests/ui-cargo/duplicate_mod/fail/src/main.rs

```diff
@@ -13,4 +15,15 @@ mod c3;
 mod from_other_module;
 mod other_module;
 
+mod d;
```

Comment:
In that case, the expectation on this and the `d3` item would be fulfilled, but no lint would be emitted for `d2`. This is because, there are no two modules, which should issue a warning. It's probably debatable, how it should behave correctly. I think this is already an improvement, over the previous implementation which ignored local attributes. :upside_down_face:

Reply:
That sounds reasonable.